### ISSUE TITLE
Enrolling iOS 17.3 doesn't work

### ISSUE THREAD

**Issue by Coder84619** [ESPresense/ESPresense#1181]
### Describe the bug

I have an iPhone 15 Pro on iOS 17.2.1, and running ESPresence 3.3.1. I click Enroll in ESPresence, and I only get a 'regular' Bluetooth pairing prompt in iOS, not a "secure" prompt. After I tap Pair in iOS, I do NOT see ANY resulting device listed in ESPresence. 

### Version

3.3.1

### Device (be specific)

iPhone 15 Pro, iOS 17.2.1

**Comment by adolfonovo** [comment]
Same here . Following

**Comment by meetbryce** [comment]
From my iPhone it looks like the pairing is successful but ESPresense doesn't seem to recognise or acknowledge it in any way. The modal stays up until it times out.

![Screenshot 2024-01-11 at 13 37 05](https://github.com/ESPresense/ESPresense/assets/1845143/a74ece97-512f-4292-a67e-2d247b6ca4c9)

**Comment by adolfonovo** [comment]
> From my iPhone it looks like the pairing is successful but ESPresense doesn't seem to recognise or acknowledge it in any way. The modal stays up until it times out.
> 
> ![Screenshot 2024-01-11 at 13 37 05](https://private-user-images.githubusercontent.com/1845143/296017066-a74ece97-512f-4292-a67e-2d247b6ca4c9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ5OTg4NTQsIm5iZiI6MTcwNDk5ODU1NCwicGF0aCI6Ii8xODQ1MTQzLzI5NjAxNzA2Ni1hNzRlY2U5Ny01MTJmLTQyOTItYTY3ZS0yZDI0N2I2Y2E0YzkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDExMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMTFUMTg0MjM0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MmY1MDI5OGE3ZjI0ZGM1MDEyM2UwN2I3MWU1ZDI4YTcxMjcyOGIzZjgyZjcyZDM5YWQzZTYyM2M3ZWJhZGYyMyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.uqLeG7z9Lrthw3fLalK6qNYXJI7NAgg-hsYtW8InHUE)

Yes! Same behavior. I also tried with the Apple Watch with out success

**Comment by meetbryce** [comment]
@adolfonovo I was able to get it working by following the instructions [here](https://espresense.com/beacons/apple) under `Lookup Method (requires a Mac)`

I figured out which one it was based on Date Modified matching the date I got my phone.

![Screenshot 2024-01-11 at 14 13 50](https://github.com/ESPresense/ESPresense/assets/1845143/231a94ef-360f-4e34-9b9c-f15ab54160fb)

Then I had to set up the sensor definition as shown below based on what I found [here](https://github.com/ESPresense/ESPresense/issues/1031#issuecomment-1780541169):

```yaml
- platform: mqtt_room
    name: "Bryce iPhone BLE"
    device_id: "irk:xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
    state_topic: "espresense/devices/irk:xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
    timeout: 5
    away_timeout: 180
```

**Comment by meetbryce** [comment]
Good to know! That's going to be become a problem for me in a few hours then when I try to add a 2nd device with a separate iCloud =(

**Comment by adolfonovo** [comment]
Thans for the info…. And what do you think could be the reason? iOS 17? ESPresense bug?
I guess the first one because i tried the enroll feature with an android device and works perfectly

**Comment by Coder84619** [comment]
@adolfonovo But I don't think Android phones use the randomized BT MACs with IRKs? I have no clue where the bug is, but it's clearly broken.

**Comment by meetbryce** [comment]
I wasn't able to connect an iPhone 14 Pro though the UI, so I suspect it's iOS 17 related.

**Comment by adolfonovo** [comment]
> I wasn't able to connect an iPhone 14 Pro though the UI, so I suspect it's iOS 17 related.

Yes, almost certainly. Let's hope the developers can overcome that block.

**Comment by meetbryce** [comment]
The workaround I shared above worked fine though, even with a different iCloud account. They just had to use their mac to get the `irk` for me.

**Comment by Coder84619** [comment]
@meetbryce But I don't see a way to configure a device name? So the enrolled devices list blank, even if the IRKs are used to decode the MACs.

**Comment by dauer1205** [comment]
Same issue with iphone 14 pro on ios 17.3

**Comment by Cyberbeni** [comment]
I've connected to ESPresense with the [nRF Connect app](https://apps.apple.com/us/app/nrf-connect-for-mobile/id1054362403) that was mentioned in the [original PR](https://github.com/ESPresense/ESPresense/pull/553) and it seems that it properly paired with my phone. (Although I had to also restart ESPresense because it went back to using the "apple:100x:..." id after like half a minute the first time)

**Comment by Cyberbeni** [comment]
Did anyone try this method for getting the irk? https://community.home-assistant.io/t/guide-getting-the-irk-for-iphone-and-apple-watch-without-macos-maybe-also-other-devices/623694

The code seems a bit different, so it might work.

**Comment by jtowe1** [comment]
i'm having the same problem, the countdown stops.  It looks like the websocket is disconnecting

![image](https://github.com/ESPresense/ESPresense/assets/62212326/2aa680b7-0989-474f-a737-429d842c6dfd)

**Comment by Cyberbeni** [comment]
> Did anyone try this method for getting the irk? https://community.home-assistant.io/t/guide-getting-the-irk-for-iphone-and-apple-watch-without-macos-maybe-also-other-devices/623694
> 
> The code seems a bit different, so it might work.

I had time to try this out and it worked 2 out of 2 times with iOS 17.3.1 devices.

FYI @DTTerastar

**Comment by DrShivang** [comment]
+1 Iphone 13 Pro with IOS 16.5.1, the enrol popup times out with nothing in messages too

**Comment by bzumik1** [comment]
Same issue here as well.

**Comment by luzik** [comment]
Yeah Enrollment do not work with IOS 17.4.1 - iPhone 12 here and Apple Watch

**Comment by bzumik1** [comment]
Did someone had a chance to test it on earlier version of iOS? Is it really issue with the latest SW.

**Comment by jtowe1** [comment]
I got a different esp board and it worked.

**Comment by luzik** [comment]
I am using esp32-s3-devkitc-1

**Comment by jtowe1** [comment]
The one I had the problem with was a 
```
LILYGO T-Display-S3 ESP32-S3 1.9 inch ST7789 LCD Display TTGO Development Board
```

The one that works for me is
```
ESP-WROOM-32 ESP32 ESP-32S Development Board 2.4GHz Dual-Mode WiFi + Bluetooth Dual Cores Microcontroller Processor Integrated with Antenna RF AMP Filter AP STA Compatible with Arduino IDE (3PCS)
```

**Comment by bzumik1** [comment]
I am having issue on **esp32 c3 super mini**.

**Comment by WilliamBradshaw** [comment]
Also having the same issue on a esp32 c3 super mini

**Comment by trollix** [comment]
Same Issue here with iphone 15

![screenshot_1625](https://github.com/ESPresense/ESPresense/assets/3890545/033a37bf-b776-443f-a6ad-5a2fb216516f)

**Comment by tunip** [comment]
Same issue on M5Stack C3U (esp32c3).

**Comment by labrunning** [comment]
I'm also having problems enrolling my iPhone. 

Beacon: iPhone 15 Plus, iOS Version 17.4.1
Base Station: EST-ESP32-S3-DevKit with ESP32-S3-WROOM-1-N8R8 (x 3)
ESPresense Version: v3.3.2

The ESPresence beacons are working as I would expect, and with a considerable amount of testing I can get the name of my phone as an id (i.e. apple:iphone15-5) but I'm concerned this is not going to be permanent. I don't have a MAC to get an irk.

**Comment by tunip** [comment]
@labrunning you can use temporary a "normal" esp32 (not a s3 or c3) for pairing your iPhone to get the IRK. Afterwards you can switch it off and/or remove the temporary ESP32 again.

**Comment by labrunning** [comment]
@tunip you like these? [https://shop.everythingsmart.io/products/est-esp32-devkit-esp32-development-kit](https://shop.everythingsmart.io/products/est-esp32-devkit-esp32-development-kit) I bought the other ones which were slightly more expensive because it expressly said they were ideal for ESPresence, turns out this isn't actually the case. I also struggled to find any kind of enclosure for them that I could buy, so I may just switch to the regular ESP32s and see if I can find another project for the S3s.

**Comment by tunip** [comment]
@labrunning yes, the ESP32-WROOM-32 based ones would work fine with ESPresense and for pairing with your iPhone. Hopefully the "pairing bug" with the S3/C3 devices are fixed in the future. After pairing, you can continue to use your ESP32-S3 device, as this is only a one-off task and the S3/C3 devices otherwise work great.

**Comment by labrunning** [comment]
@tunip thanks I will give that a try.

**Comment by uohlukas** [comment]
I've tried with the "standard" ESP32 and managed to successfully pair my iPhone and now have a working irk number which has made everything much easier. Thanks for the help.

**Comment by labrunning** [comment]
I flashed espresence on to a "standard" ESP32 today and managed to successfully enrol my IPhone. I now have an irk number which is making things much easier. Thanks for the help.

**Comment by Tuinslak** [comment]
> ESP32-WROOM-32



> this is only a one-off task and the S3/C3 devices otherwise work great.

Hey -- so that I understand correctly (as a new user; as I haven't been able to get any further): the pairing only has to happen once (on one device) across the entire network of ESPresence? It's not something that needs to be done on every individual board? Is it a manual process of copying the IRK across devices, or is automatic?

**Comment by tunip** [comment]
> Hey -- so that I understand correctly (as a new user; as I haven't been able to get any further): the pairing only has to happen once (on one device) across the entire network of ESPresence? It's not something that needs to be done on every individual board? Is it a manual process of copying the IRK across devices, or is automatic?

You are right. Pairing only has to happen once on one device across the entire network of ESPresense. It is only to get the IRK from your iPhone/iPad which is "stored" on your MQTT Broker. Each ESPresense node can then "read" this information via MQTT.

**Comment by lachi3** [comment]
This may have already been solved above. But i have an old iPad that isn't on iOS 17, and i'm still getting the same result.

**Comment by broyuken** [comment]
I'm unable to get the irk for my phones or watches on both ios17 and 18 beta.

**Comment by raptile** [comment]
same here on with multiple iphones.

**Comment by schmiegelt** [comment]
Same here, doesn't work with iOS 17.
Unfortunatly I do not have a Mac available, so I cannot directly use the alternative method of retrieving the IRK. Is there any service where you can spin up a Mac VM for an hour or two? The one's I found are rather expensive and not suitable for short-term usage.

**Comment by slashinpdx** [comment]
Having the same problem with the iPhone 16 Pro.  Unable to use the mac lookup method as well.  :(

**Comment by akcoder** [comment]
Same issues as everyone else. I was able to get both my watch and phone enrolled, but I had to downgrade one of my nodes to **v3.2.4**.

Device: ESP32-S3

**Comment by yuriisk** [comment]
Same issue.

Was able to get irk using friend's mac. To do that with different iCloud account I needed to pair iphone from mac bluetooth settings and then extract irk from Keychain Access.

**Comment by drax16** [comment]
I know it's late, but I'm still having the same issue @meetbryce had. It appears to be connected on my iPhone itself, but it refuses to enroll on the web-ui side of things. I have a mac, but I want to add other devices that are not connected to my Apple ID. Using an ESP32 C3

**Comment by quietCoder500** [comment]
I am also having the exact same issue, but I have no access to a mac to get the irk.

**Comment by mattiasa** [comment]
After updating one of my ESP32-C3 to ESPresence 4.0.0b11 I was able to successfully pair an iPhone running iOS 18.6.2

**Comment by dverbeir** [comment]
I can confirm v4.0.0b11 worked on ESP32-C3 to enroll an iPhone 15 running iOS 26.0.1. It didn't work with the v3.3.5 release.
